sdk/formrecognizer/azure-ai-formrecognizer/src/FormRecognizer.md

```diff
@@ -0,0 +1,501 @@
+## Form recognizer API design proposal
+
+### Methods
+
+```java
+// Train
+public PollerFlux<OperationResult, Model> beginModelTraining(String sourceUrl) { }
+public PollerFlux<OperationResult, Model> beginModelTraining(String sourceUrl, String filePrefix, boolean includeSubFolders) { }
+
+public PollerFlux<OperationResult, SupervisedModel> beginSupervisedModelTraining(String sourceUrl) { }
+public PollerFlux<OperationResult, SupervisedModel> beginSupervisedModelTraining(String sourceUrl, String filePrefix, boolean includeSubFolders) { }
+
+// Analyze form with custom models
+public PollerFlux<OperationResult, FormResult> beginAnalyzeForm(Flux<ByteBuffer> data, String modelId, boolean includeTextDetails) { }
+public PollerFlux<OperationResult, FormResult> beginAnalyzeForm(String uploadFilePath, String modelId, boolean includeTextDetails) { }
+
+// Analyze with supervised models
+public PollerFlux<OperationResult, PredefinedFormResult> beginSupervisedAnalyzeForm(Flux<ByteBuffer> data, String modelId, boolean includeTextDetails) { }
+public PollerFlux<OperationResult, PredefinedFormResult> beginSupervisedAnalyzeForm(String uploadFilePath, String modelId, boolean includeTextDetails) { }
+
+// List custom Models
+public PagedFlux<ModelInfo> listModels() {}
+public ModelSummary getModelListSummary() {}
+public Response<ModelSummary> getModelListSummaryWithResponse() {}
+
+// Delete model
+public Mono<Void> delete(String modelId) { }
+public Mono<Response<Void>> deleteWithResponse(String modelId) { }
+
+// Analyze receipt 
+public PollerFlux<OperationResult, ExtractedReceiptResult> beginAnalyzeReceipt(Flux<ByteBuffer> data, boolean includeTextDetails) { }
```

Comment:
What exactly are the issues of hot vs cold flux that we are seeing in storage? This would help to know if they would apply here as well.

Reply:
Given that the data expected here should be of a certain size and type specification, would using a simple InputStream suffice?

Reply:
Deferring this to @JonathanGiles for guidance.

Reply:
So, yes, InputStream would suffice for the use case, however, the intent of the async API is to be non-blocking, which is why we try to use Flux<ByteBuffer>. I think this needs further investigation.

Reply:
(and I encourage you to do it, because I do worry about the complexity of this API - maybe there might be value in either overloads or clear documentation on how to transform an InputStream into this, etc).